### PR TITLE
dynamic NPM package installer utility

### PR DESCRIPTION
**Pull request by jpw**
```
This package allows dynamic installation of other NPM packages.

Internally it uses node's `child_process.spawn` because the recommended way of
installing node modules dynamically is to use `npm` directly.

It has one dependency and is intended to be as light as possible.
```
Quite a bit going on in here for a relatively simple module, feel free to ask questions ;)

We need this for the forthcoming utility that renders a component inline, and some other bits of the publishing system.

It includes a `.gitignore` etc. entry for a `docs` folder, because quite nice JSDocs can be generated from the module. However the docs are not included in this PR, to make it easier to review.

EDIT: This was crashing `npm` on CI due to excessive memory usage of Jest, apparently when generating the coverage report. It seems fine today, so assume Travis was low on resources. To help mitigate this in the future, have added:
```
"test:ci": "jest --passWithNoTests --runInBand"
```
...which stops running coverage on CI, and adds `--runInBand` which is supposed to help Jest's high memory usage.

### CODE REVIEW

File: packages/util-package-installer/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+The general idea is to pass it a `JSON.parse`d `package.json`;
+It will either return `stdout`, or throw an `Error`.
+
+```
+const install = require('@springernature/util-package-installer');
+const packageJSON = require('./path/to/package.json');
+(async () => {
+    let installResult;
+    try {
+        installResult = await install.peerDependencies(packageJSON);
```
Comment (alexkilgour) [inline packages/util-package-installer/README.md]: When I read this, I thought the only option was to `install.peerDependencies`. Maybe worth noting above this example which methods are actually exported - `install.dependencies`, `install.devDependencies` etc...

File: packages/util-package-installer/HISTORY.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# History
+
+## 0.0.1 (2019-09-24)
+    * Initial release
```
Comment (jon-stevens) [inline packages/util-package-installer/HISTORY.md]: does this need to be 0.1.0 to match package.json?

**Comment by jon-stevens** [comment]
Looks super to me

**Comment by jpw** [comment]
Thanks gents!

**Pull request merged**